tests/apps/demultiplex/test_index.py

```diff
@@ -308,5 +308,37 @@ def test_get_hamming_distance_index_different_prefixes():
     # WHEN getting the hamming distance between them in any order
 
     # THEN the distance is greater than zero
-    assert get_hamming_distance_for_indexes(sequence_1=sequence_1, sequence_2=sequence_2) > 0
-    assert get_hamming_distance_for_indexes(sequence_1=sequence_2, sequence_2=sequence_1) > 0
+    assert get_hamming_distance_index_1(sequence_1=sequence_1, sequence_2=sequence_2) > 0
+    assert get_hamming_distance_index_1(sequence_1=sequence_2, sequence_2=sequence_1) > 0
+
+
+def test_get_hamming_distance_index_2_different_lengths():
+    """Test that the hamming distance between indexes with same suffix but different lengths is zero."""
+    # GIVEN two index_2 sequences with the same suffixes but different lengths
+    sequence_1: str = "GATTACA"
+    sequence_2: str = "XXGATTACA"
+
+    # WHEN getting the hamming distance between them in any order
+
+    # THEN the distance is zero
+    assert get_hamming_distance_index_2(sequence_1=sequence_1, sequence_2=sequence_2) == 0
+    assert get_hamming_distance_index_2(sequence_1=sequence_2, sequence_2=sequence_1) == 0
+
+    # WHEN getting the hamming distance between themselves
+
+    # THEN the distance is zero
+    assert get_hamming_distance_index_2(sequence_1=sequence_1, sequence_2=sequence_1) == 0
+    assert get_hamming_distance_index_2(sequence_1=sequence_2, sequence_2=sequence_2) == 0
+
+
+def test_get_hamming_distance_index_2_different_prefixes():
+    """Test that the hamming distance between indexes with different suffixes is greater than zero."""
+    # GIVEN two index_2 sequences with the same prefixes but different suffixes
+    sequence_1: str = "GATTACA"
+    sequence_2: str = "GATTACAXX"
+
+    # WHEN getting the hamming distance between them in any order
+
+    # THEN the distance is greater than zero
+    assert get_hamming_distance_index_2(sequence_1=sequence_1, sequence_2=sequence_2) > 0
+    assert get_hamming_distance_index_2(sequence_1=sequence_2, sequence_2=sequence_1) > 0
```

Comment:
Any reason to not assert this to be the exact number that you expect it to be? I feel like this is something we cannot accept be ANYTHING but 2. Goes for the tests above as well.

Reply:
In this case, the distance is actually not 2 but 6. As this is an index2 comparison, it will always compare aligning from the tail. Making the expected distance explicitly equal to 6 in my opinion, is not that meaningful as you as a reviewer would have to manually verify that aligning by the tails gives 6, while the most important thing is to show that the function is not aligning from the head, otherwise it would be zero. I can put it as 6 if you think it would be clearer but honestly, I think the number is not important.

Reply:
Ok I refactored the test to be clearer